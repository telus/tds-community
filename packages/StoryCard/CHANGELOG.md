# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.1.0](https://github.com/telus/tds-community/compare/@tds/community-story-card@1.0.3...@tds/community-story-card@1.1.0) (2021-09-21)


### Features

* **community-story-card:** react 17 and styled components 5 ([a139b9a](https://github.com/telus/tds-community/commit/a139b9a789b1a2d26dcf9ff525380a6b07bd303a))





## [1.0.3](https://github.com/telus/tds-community/compare/@tds/community-story-card@1.0.2...@tds/community-story-card@1.0.3) (2021-07-20)


### Bug Fixes

* **community-story-card:** update height styling and remove div ([dbd3bc2](https://github.com/telus/tds-community/commit/dbd3bc2483d0e4fc93f09d642efc4061838e0c75))





## [1.0.2](https://github.com/telus/tds-community/compare/@tds/community-story-card@1.0.1...@tds/community-story-card@1.0.2) (2021-06-30)


### Bug Fixes

* **community-story-card:** update util helper version ([9cdbfa9](https://github.com/telus/tds-community/commit/9cdbfa98b6c6174690a93ec058ab61d74857ed29))





## [1.0.1](https://github.com/telus/tds-community/compare/@tds/community-story-card@1.0.0...@tds/community-story-card@1.0.1) (2021-06-29)


### Bug Fixes

* **community-story-card:** add media queries ([ba19936](https://github.com/telus/tds-community/commit/ba19936b830bc82259ad5792bffbd94733c4f999))





# 1.0.0 (2021-06-22)


### Features

* **community-story-card:** add functionality and design for storycard ([85a2a84](https://github.com/telus/tds-community/commit/85a2a84b363037bc7d1971a64958de1cb5d77db1))
