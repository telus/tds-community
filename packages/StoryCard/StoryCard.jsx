import React from 'react'
import PropTypes from 'prop-types'
import { News, Headset, Play, Clipboard } from '@tds/core-decorative-icon'
import {
  StyledCard,
  StyledTextBox,
  StyledInfoBox,
  StyledType,
  StyledDate,
  StyledHeadingBox,
  StyledHeading,
  StyledDescription,
  StyledIconTypeBox,
  StyledLink,
  StyledIconBox,
  StyledImageContainer,
  StyledImage,
} from './style'

const StoryCard = ({ storyType, date, title, description, imgUrl, slug }) => {
  const selectIcon = type => {
    const normalizedType = type.toLowerCase()
    if (normalizedType === 'article') {
      return <News size="20" />
    } else if (normalizedType === 'podcast') {
      return <Headset size="20" />
    } else if (normalizedType === 'video') {
      return <Play size="20" />
    } else if (normalizedType === 'case study') {
      return <Clipboard size="20" />
    }
    return undefined
  }

  return (
    <StyledLink href={slug}>
      <StyledCard>
        <StyledTextBox>
          <StyledInfoBox>
            <StyledIconTypeBox>
              {selectIcon(storyType) !== undefined && (
                <StyledIconBox>{selectIcon(storyType)}</StyledIconBox>
              )}
              <StyledType>{storyType}</StyledType>
            </StyledIconTypeBox>
            <StyledDate>{date}</StyledDate>
          </StyledInfoBox>
          <StyledHeadingBox>
            <StyledHeading>{title}</StyledHeading>
          </StyledHeadingBox>
          <StyledDescription>{description}</StyledDescription>
        </StyledTextBox>
        <StyledImageContainer>
          <StyledImage src={imgUrl} alt={description} width="100%" />
        </StyledImageContainer>
      </StyledCard>
    </StyledLink>
  )
}

StoryCard.propTypes = {
  /**
   * The title of the story
   */
  title: PropTypes.string,
  /**
   * The description of the story
   */
  description: PropTypes.string,
  /**
   * The URL for the image to be displayed
   */
  imgUrl: PropTypes.string,
  /**
   * The slug for the story
   */
  slug: PropTypes.string,
  /**
   * The type of story, if it is one of Article, Podcast or Video, an icon will render beside it. If it is something else, there will be no icon.
   */
  storyType: PropTypes.string,
  /**
   * The date of when the story was published
   */
  date: PropTypes.string,
}

StoryCard.defaultProps = {
  title: undefined,
  description: undefined,
  imgUrl: undefined,
  slug: undefined,
  storyType: undefined,
  date: undefined,
}

export default StoryCard
