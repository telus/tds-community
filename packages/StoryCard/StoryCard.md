The StoryCard component is meant to display a snippet of an Article, Podcast or Video. The whole card is clickable, to take you to the destination of said story.

### Usage criteria

- Typical usage of this component will be to display a snippet of an Article, Podcast or Video
- This component is completely dynamic, so it can likely be modified to suit other needs

```jsx
<FlexGrid>
  <FlexGrid.Col xs={12} md={6}>
    <StoryCard
      storyType="Article"
      date="May 11th, 2021"
      title="Crisis Text Line provides mental health support for youth during lockdown"
      description="With the help of a Foundation grant, Kids Help Line was able to expand its mental health resources to include a Crisis Text Line in greater Edmonton."
      imgUrl="blog-example.jpg"
      slug="kids-help-line"
    />
  </FlexGrid.Col>
</FlexGrid>
```
