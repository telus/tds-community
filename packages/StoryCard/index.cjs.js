const StoryCard = require('./dist/index.cjs')

module.exports = StoryCard
