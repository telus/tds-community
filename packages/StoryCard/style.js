import styled from 'styled-components'
import { colorGreyGainsboro, colorTelusPurple, colorWhite } from '@tds/core-colours'

export const StyledCard = styled.div`
  max-width: 29.188rem;
  min-height: 38.875rem;
  background: ${colorWhite};
  border-radius: 3px;
  border: 1px solid ${colorGreyGainsboro};
  box-shadow: 0 0 1rem #d8d8d8;
  transition: transform 0.4s;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  &:hover {
    transform: scale(103%);
    border: 0.125rem solid ${colorTelusPurple};
  }
`

export const StyledLink = styled.a`
  text-decoration: none;
  color: inherit;
`

export const StyledTextBox = styled.div`
  padding: 1.5rem 1rem;
`

export const StyledInfoBox = styled.div`
  display: flex;
  justify-content: space-between;
`

export const StyledType = styled.p`
  color: ${colorTelusPurple};
  font-size: 0.875rem;
`

export const StyledDate = styled.p`
  font-size: 0.875rem;
`

export const StyledHeading = styled.h2`
  font-size: 1.219rem;
  font-weight: 400;
  line-height: 1.4;
`

export const StyledHeadingBox = styled.div`
  padding: 1.5rem 0;
`

export const StyledIconBox = styled.div`
  padding-right: 0.5rem;
`

export const StyledImageContainer = styled.div``

export const StyledDescription = styled.p`
  font-size: 0.875rem;
`

export const StyledIconTypeBox = styled.div`
  display: flex;
  border-bottom: 1px solid ${colorTelusPurple};
  padding-bottom: 8px;
`

export const StyledImage = styled.img`
  display: block;
  border-radius: 0 0 2px 2px;
`
