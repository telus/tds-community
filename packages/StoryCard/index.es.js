import StoryCard from './dist/index.es'

export default StoryCard
