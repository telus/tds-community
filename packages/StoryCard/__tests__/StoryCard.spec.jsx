import React from 'react'
import { shallow, mount } from 'enzyme'

import StoryCard from '../StoryCard'

describe('StoryCard', () => {
  const defaultProps = {
    storyType: 'Article',
    date: 'May 11th, 2021',
    title: 'Crisis Text Line provides mental health support for youth during lockdown',
    description: 'With the help of a Foundation grant',
    imgUrl: 'blog-example.jpg',
    slug: 'kids-help-line',
  }
  const doShallow = () => shallow(<StoryCard {...defaultProps} />)

  it('renders', () => {
    const storyCard = doShallow()

    expect(storyCard).toMatchSnapshot()
  })

  it('passes additional attributes to the element', () => {
    const props = {
      ...defaultProps,
      id: 'the-id',
      'data-some-attr': 'some value',
    }
    const previewCard = mount(<StoryCard {...props} />)

    expect(previewCard).toHaveProp('id', 'the-id')
    expect(previewCard).toHaveProp('data-some-attr', 'some value')
  })

  it('does not allow custom CSS', () => {
    const storyCard = doShallow({
      className: 'my-custom-class',
      style: { color: 'hotpink' },
    })

    expect(storyCard).not.toHaveProp('className', 'my-custom-class')
    expect(storyCard).not.toHaveProp('style')
  })
})
