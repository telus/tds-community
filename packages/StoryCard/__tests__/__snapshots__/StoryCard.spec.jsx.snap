// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryCard renders 1`] = `
<style__StyledLink
  href="kids-help-line"
>
  <style__StyledCard>
    <style__StyledTextBox>
      <style__StyledInfoBox>
        <style__StyledIconTypeBox>
          <style__StyledIconBox>
            <DecorativeIcon
              size="20"
            />
          </style__StyledIconBox>
          <style__StyledType>
            Article
          </style__StyledType>
        </style__StyledIconTypeBox>
        <style__StyledDate>
          May 11th, 2021
        </style__StyledDate>
      </style__StyledInfoBox>
      <style__StyledHeadingBox>
        <style__StyledHeading>
          Crisis Text Line provides mental health support for youth during lockdown
        </style__StyledHeading>
      </style__StyledHeadingBox>
      <style__StyledDescription>
        With the help of a Foundation grant
      </style__StyledDescription>
    </style__StyledTextBox>
    <style__StyledImageContainer>
      <style__StyledImage
        alt="With the help of a Foundation grant"
        src="blog-example.jpg"
        width="100%"
      />
    </style__StyledImageContainer>
  </style__StyledCard>
</style__StyledLink>
`;
